src/Sulu/Bundle/DocumentManagerBundle/Command/PHPCRCleanupCommand.php

```diff
@@ -0,0 +1,254 @@
+
+        $chunkSize = (int) $input->getOption('processes');
+        Assert::greaterThan($chunkSize, 0, 'Chunk size must be greater than 0');
+
+        $chunks = \array_chunk($uuids, $chunkSize);
+        foreach ($chunks as $chunk) {
+            $processes = [];
+
+            /** @var string $uuid */
+            foreach ($chunk as $uuid) {
+                $processes[$uuid] = $this->createProcess($uuid, $dryRun, $debug);
+                $processes[$uuid]->start();
+            }
+
+            foreach ($processes as $uuid => $process) {
+                ++$stats['nodes'];
+                $status = $process->wait();
+                if (PHPCRCleanupSingleNodeCommand::IGNORED === $status) {
+                    ++$stats['ignoredNodes'];
+                    $ignoredUuids[] = $uuid;
+
+                    continue;
+                }
+                if (0 !== $status) {
+                    ++$stats['erroredNodes'];
+                    $erroredUuids[] = $uuid;
+
+                    continue;
+                }
+
+                $output = $process->getOutput();
+                \preg_match('/Documents: (\d+)/', $output, $matches);
+                $stats['documents'] += (int) ($matches[1] ?? 0);
+                \preg_match('/Removed properties: (\d+)/', $output, $matches);
+                $stats['removedProperties'] += (int) ($matches[1] ?? 0);
+                \preg_match('/Total properties: (\d+)/', $output, $matches);
+                $stats['properties'] += (int) ($matches[1] ?? 0);
+
+                $this->logger->writeln($output);
+
+                $progressBar->setMessage((string) $stats['nodes'], 'nodes');
+                $progressBar->setMessage((string) $stats['ignoredNodes'], 'ignoredNodes');
+                $progressBar->setMessage((string) $stats['erroredNodes'], 'erroredNodes');
+                $progressBar->setMessage((string) $stats['documents'], 'documents');
+                $progressBar->setMessage((string) $stats['properties'], 'properties');
+                $progressBar->setMessage((string) $stats['removedProperties'], 'removedProperties');
+                $progressBar->advance();
+            }
+
+            $this->servicesResetter->reset();
+        }
+
+        $progressBar->finish();
+        $io->success('Cleanup process finished');
+
+        if (0 < \count($erroredUuids)) {
+            $io->section('Following nodes are errored');
+            $io->listing($erroredUuids);
+        }
+
+        if (0 < \count($ignoredUuids)) {
+            $io->section('Following nodes are ignored');
+            $io->listing($ignoredUuids);
+        }
+
+        return self::SUCCESS;
+    }
+
+    protected function createProcess(string $uuid, bool $dryRun, bool $debug): Process
+    {
+        $executableFinder = new PhpExecutableFinder();
+        $php = $executableFinder->find(false);
+
+        $process = new Process(\array_filter([
+            $php,
+            $_SERVER['argv'][0],
```

Comment:
@wachterjohannes @mamazu `$_SERVER['argv']` is risky and should not be accessed. This should be handled differently. Symfony resets for security reasons the values: https://github.com/symfony/symfony/pull/58785/files

Reply:
Isn't there a PhpProcess class in Symfony?

Reply:
yeah but that is not the issue `$_SERVER` should not be accessed.

Reply:
I agree, however I don't know if there is an easy way of getting the path of the script that is currently executing in PHP otherwise.

Reply:
`$GLOBALS['argv']` could be a workaround. or `$_SERVER['SCRIPT_NAME']`

Reply:
I mean phpcr is on the way out anyways. :D